skins/cat17/address-form/src/types.ts

```diff
@@ -1,17 +1,30 @@
-export interface FormValidity {
+export interface FieldValidity {
 	dataEntered: boolean,
 	isValid: boolean | null
 }
 
 export interface Form {
-	[key: string]: FormValidity
+	[key: string]: FieldValidity
+}
+
+export interface AddressValidity {
+	salutation: boolean,
+	companyName: boolean,
+	firstName: boolean,
+	lastName: boolean,
+	street: boolean,
+	city: boolean,
+	postcode: boolean
```

Comment:
I've done your proposed change (also with the iteration in addressForm.fieldErrors), but I can't shake the feeling that stylistically we're getting into 2000's "everything is an array"-style PHP code territory. At least the array (or rather hashmap) is typed, but still, we're losing expressiveness and explicitness in favor of conciseness. I think this case here is fine, but let's not overdo iteratable types. I like explicit statements, even when it adds verbosity.

When we add or remove fields from the address form, we need to think if putting them into `AddressValidity` makes sense from a semantic perspective.

Reply:
I know what you mean but in this case every property is `boolean` and there's no chance we add something that isn't to that particular type.  
Also, I think the reduce function does a good job of explaining why exactly these 7 out of our 10 fields need to be validated. Since we have `formData` explicitly describing all the fields and their properties I don't feel like we need to repeat that in the types.
I can imagine someone who was not part of building this having to add one more field to the address form. They have to figure out what to add and if their new field should be added at all in all three places. It's not a big deal but I still think we should avoid it.
I did not request changes because I was unsure if the combination of iterating over Object keys and using `reduce` is worth it over repeating field names 3 times. In my opinion it is, but it is just my opinion/suggestion.
Anyway, thanks for addressing it!